src/set_theory/surreal.lean

```diff
@@ -141,47 +128,136 @@ begin
   let y := mk yl yr yL yR,
   refine ⟨equiv.sum_congr (equiv.prod_comm _ _) (equiv.prod_comm _ _), _, _, _⟩,
   calc
-   (x * y).right_moves
-       ≃ xl × yr ⊕ xr × yl : by refl
-   ... ≃ xr × yl ⊕ xl × yr : equiv.sum_comm _ _
-   ... ≃ yl × xr ⊕ yr × xl : equiv.sum_congr (equiv.prod_comm _ _) (equiv.prod_comm _ _)
-   ... ≃ (y * x).right_moves : by refl,
+    xl × yr ⊕ xr × yl
+       ≃ xr × yl ⊕ xl × yr : equiv.sum_comm _ _
+   ... ≃ yl × xr ⊕ yr × xl : equiv.sum_congr (equiv.prod_comm _ _) (equiv.prod_comm _ _),
   { rintro (⟨i, j⟩ | ⟨i, j⟩),
-    { exact add_comm_sub_relabelling (IHxl i y) (IHyl j) (IHxl i (yL j)) },
-    { exact add_comm_sub_relabelling (IHxr i y) (IHyr j) (IHxr i (yR j)) }},
+    { exact ((add_comm_relabelling _ _).trans
+                ((IHyl j).add_congr (IHxl i y))).add_congr (IHxl i (yL j)).neg_congr },
+    { exact ((add_comm_relabelling _ _).trans
+                ((IHyr j).add_congr (IHxr i y))).add_congr (IHxr i (yR j)).neg_congr } },
   { rintro (⟨i, j⟩ | ⟨i, j⟩),
-    { exact add_comm_sub_relabelling (IHxr j y) (IHyl i) (IHxr j (yL i)) },
-    { exact add_comm_sub_relabelling (IHxl j y) (IHyr i) (IHxl j (yR i)) }}
+    { exact ((add_comm_relabelling _ _).trans
+                ((IHyl i).add_congr (IHxr j y))).add_congr (IHxr j (yL i)).neg_congr },
+    { exact ((add_comm_relabelling _ _).trans
+                ((IHyr i).add_congr (IHxl j y))).add_congr (IHxl j (yR i)).neg_congr } }
 end
 
 /-- `x * y` is equivalent to `y * x`. -/
-theorem mul_comm_equiv (x y : pgame) : (x * y).equiv (y * x) :=
+theorem mul_comm_equiv (x y : pgame) : x * y ≈ y * x :=
 (mul_comm_relabelling x y).equiv
 
 /-- `x * 0` has exactly the same moves as `0`. -/
 def mul_zero_relabelling : Π (x : pgame), relabelling (x * 0) 0
 | (mk xl xr xL xR) :=
-⟨by fsplit; rintro (⟨_,⟨⟩⟩ | ⟨_,⟨⟩⟩),
- by fsplit; rintro (⟨_,⟨⟩⟩ | ⟨_,⟨⟩⟩),
- by rintro (⟨_,⟨⟩⟩ | ⟨_,⟨⟩⟩),
+⟨by fsplit; rintro (⟨_,⟨⟩⟩|⟨_,⟨⟩⟩),
+ by fsplit; rintro (⟨_,⟨⟩⟩|⟨_,⟨⟩⟩),
+ by rintro (⟨_,⟨⟩⟩|⟨_,⟨⟩⟩),
  by rintro ⟨⟩⟩
 
 /-- `x * 0` is equivalent to `0`. -/
-theorem mul_zero_equiv (x : pgame) : (x * 0).equiv 0 :=
+theorem mul_zero_equiv (x : pgame) : x * 0 ≈ 0 :=
 (mul_zero_relabelling x).equiv
 
 /-- `0 * x` has exactly the same moves as `0`. -/
 def zero_mul_relabelling : Π (x : pgame), relabelling (0 * x) 0
 | (mk xl xr xL xR) :=
-⟨by fsplit; rintro (⟨⟨⟩,_⟩ | ⟨⟨⟩,_⟩),
- by fsplit; rintro (⟨⟨⟩,_⟩ | ⟨⟨⟩,_⟩),
- by rintro (⟨⟨⟩,_⟩ | ⟨⟨⟩,_⟩),
+⟨by fsplit; rintro (⟨⟨⟩,_⟩|⟨⟨⟩,_⟩),
+ by fsplit; rintro (⟨⟨⟩,_⟩|⟨⟨⟩,_⟩),
+ by rintro (⟨⟨⟩,_⟩|⟨⟨⟩,_⟩),
  by rintro ⟨⟩⟩
 
 /-- `0 * x` is equivalent to `0`. -/
-theorem zero_mul_equiv (x : pgame) : (0 * x).equiv 0 :=
+theorem zero_mul_equiv (x : pgame) : 0 * x ≈ 0 :=
 (zero_mul_relabelling x).equiv
 
+lemma left_distrib_equiv_aux {a b c d e : pgame} : a + b + (c + d) - (e + b) ≈ a + c - e + d :=
+begin
+  apply @quotient.exact pgame,
+  change (⟦a⟧ + ⟦b⟧ + (⟦c⟧ + ⟦d⟧) - (⟦e⟧ + ⟦b⟧) : game) = ⟦a⟧ + ⟦c⟧ + -⟦e⟧ + ⟦d⟧,
```

Comment:
Is there a `simp` lemma that can achieve this `change`? Whether or not you actually use it, there probably should be one.

Reply:
Currently there is none. If I try to define something like the following in game.lean

```lean
@[simp] lemma test (a b : pgame) : ⟦a + b⟧ = ⟦a⟧ + ⟦b⟧ := sorry
```

I'm getting an error:

```lean
141:50: failed to synthesize type class instance for
a b : pgame
⊢ has_add (quotient pgame.setoid)
```

This is a bit weird as `game` is defined as `quotient pgame.setoid`.

I can leave a TODO note in game.lean and add these lemmas later.

Reply:
Figured this out!